gpu/vulkan/features.gni

```diff
@@ -10,6 +10,7 @@ declare_args() {
   # Enable experimental vulkan backend.
   enable_vulkan =
       is_linux || is_chromeos || is_android || is_fuchsia || is_win || is_mac
+      && !use_starboard
```

Comment:
Could we put this in a share `*.gni` file that gets pulled in for all Starboard configurations ? I'd rather not change this in upstream code if possible.

CC @andrewsavage1 

Also, somewhere in a comment or such, i'd link back to our Technical Requirements stating OpenGL ES 2.0 as our base graphics API.

Reply:
The expression should be !is_starboard, right?
Also for ATV using Vulkan could be fine if the device supports it.

Also following the existing pattern and just adding is_starboard in this file looks good to me.

Reply:
You'll need to change this to be (is_foo || is_bar) && !use_starboard, right now I don't think it's doing anything. Also I agree that it would be better in a build_overrides or gn.py if possible, but we can follow up with that

Reply:
Just a note that I investigated a bit //build_override and my  
understanding is that it's only applicable to //third_party  
folder ([1]). IOW I don't think we could have a e.g. `starboard`  
entry there (even though it would have sounded amazing). 


[1] https://source.chromium.org/chromium/chromium/src/+/main:build_overrides/

Reply:
I tried incorporating all this feedback into the PR here: https://github.com/youtube/cobalt/pull/4375 ptal